File: C++/factorypattern/Tester.cpp
#include "iostream"
#include "string"

#include "./factory/ShapeFactory.h"

using namespace std;

int main() {

    ShapeFactory* shapeFactory = new ShapeFactory();
    Shape* shape = shapeFactory->getShape("CIRCLE");
    shape->draw();

    shape = shapeFactory->getShape("SQUARE");
    shape->draw();

    shape = shapeFactory->getShape("PYRAMID");
    shape->draw();

    return 0;
}

File: Java/factorypattern/Tester.java
package factorypattern;

public class Tester {
    public static void main(String[] args) {
        String shapeType = "square";

        Shape shape = null;
        if(shapeType.equals("circle")) {
            shape = new Circle();
        } else if(shapeType.equals("square")) {
            shape = new Square();
        } else if(shapeType.equals("rectangle")) {
            shape = new Rectangle();
        }

        shape.getArea();
    }
}
